### ISSUE TITLE
BUG: Broken looping Sounds

### ISSUE THREAD

**Issue by jkr4m3r** [QW-Group/ezquake-source#972]
Please search for existing issues and check for potential duplicates before filing yours.

**ezQuake version:**3.6.5

**OS/device including version:**
Win10

**Describe the bug**
Ambient sounds are no longer loading or playing. Doesn't matter whether they are new or the ambient sounds from the mod I play, still using the original .pak files from 1997.

**To Reproduce**
Add ambient looping wind or mod sound effects to a map, only players with earlier versions of ezquake hear the sounds.

**Expected behavior**
Expected behavior is that ambient sounds that loop in a 25-year-old game should work as they have for that long. Last version they work in is 3.2.3 AFAIK.

**Screenshots**
Zipped cfg and screenshot attached.

![looped](https://github.com/user-attachments/assets/fe18b071-44cc-4575-966d-2876f4f13b2c)
[config.zip](https://github.com/user-attachments/files/18053974/config.zip)

**Comment by dsvensson** [comment]
I successfully produced some looping sounds by crafting wav headers like this a while back. Think I just checked the vanilla id1 sounds and did the same thing. I did this to help out a fellow mapper who had working looping sounds in other engines that didn't loop in ezQuake.

https://github.com/qw-ctf/ambiencefix

Might be that other engines support a less archaic version of loop headers. 

But there has been a few changes to [snd_mem.c ](https://github.com/QW-Group/ezquake-source/commits/97a9b1884d03796cec15043d2c6e62ab93694389/snd_mem.c?browsing_rename_history=true&new_path=src/snd_mem.c&original_branch=master)since 3.2.3, you could always have a look at those and see if you find something if you don't want to use ambiencefix to generate queue headers.

**Comment by jkr4m3r** [comment]
> you could always have a look at those and see if you find something if you don't want to use ambiencefix to generate cue headers.

I'm afraid that's way over my head! So is the ambiencefix thing... Win10 here
Looked like Linux commands?

**Comment by dsvensson** [comment]
Nah, it's go, so you can just build it from the command line if you have go installed, it's platform independent. But if you're not creating new stuff, it's less relevant. Nice to know that 3.2.3 worked. Devs around that time - that are gone now - seems to have replaced the playback mechanism, so very much a bug that someone needs to chase down.

**Comment by jkr4m3r** [comment]
That's my point. I have no clue what "go" is. Not a developer. I tinker with weapon strengths in Quake C here and there and fiddle with maps for a certain mod, beyond that it's all over my head :P

**Comment by dsvensson** [comment]
It's not. Any google search or AI prompt ought to be able to provide you with extremely simple step-by-step instructions on how to install go and compile a single go file without any prior knowledge. The question is more if you want to do it, or if you want to gamble on the bug being fixed. Don't build fake limits on what you can achieve, it's 2024, halfway to the future.

**Comment by jkr4m3r** [comment]
I'm 65.

**Comment by dsvensson** [comment]
Stop flexing with your infinite amount of spare time 😂

**Comment by jkr4m3r** [comment]
That's the issue, always wondering why I got up and came into this room. I'll remember eventually.

**Comment by hemostx** [comment]
> This issue looks related, #671

For sure related. In fact I checked this thread hoping it would be referenced.
Would have to go back to pre https://github.com/QW-Group/ezquake-source/commit/95f4f98203be115994ef82fe12c8eb1cd6a31a5e (but not all the way to 3.2x - and rebuild without vcpkg, yuk !) to check playback.
Would also need to procure the painkeep paks and figure out which ones are looped so someone can reproduce with them and see how different from #671 it is (if at all).

**Comment by jkr4m3r** [comment]
All ambient sounds played fine in 3.2.3, is there some reason why this can't be fixed so they work in newer versions? Without having to redo the .wavs, I mean.

FWIW, I prefer 3.2.3 for various reasons, and will eventually move on to newer versions. But a couple of guys I play with are using 3.6.5 now and always see the errors in the console on map load and never hear the ambient sounds (which are cool in Painkeep).

**Comment by jkr4m3r** [comment]
Dang, I assumed by now the recent ezquake builds would have fixed sounds. Am I being foolish to expect this issue to simply get fixed as ezquake progresses?

**Comment by dsvensson** [comment]
Oh, already mentioned in thread. Anyways, use that or get someone to fix it.

**Comment by jkr4m3r** [comment]
Thanks, way over my head, I was hoping the ezquake devs would include a pak with proper ambient sounds in newer builds. They haven't. 

Also, when you say "if you loop the same way as idSoftware did back in the day it works", that sounds like my vanilla quake sounds in my original pak should be fine, but as far as I can see, they are broke. Am I missing something obvious here? (wouldnt be surprising)

**Comment by dsvensson** [comment]
> Dang, I assumed by now the recent ezquake builds would have fixed sounds. Am I being foolish to expect this issue to simply get fixed as ezquake progresses?

I think if it was widespread it would have been fixed by now.

> But a couple of guys I play with are using 3.6.5 now and always see the errors in the console

I assume these errors are irrelevant, or they would have been posted here?

> on map load and never hear the ambient sounds

Teleporters use looping sounds, and they play just fine, as does water, wind etc etc. Perhaps you have some weird files hiding in some pak/pk3 that override the idSoftware sounds?

> This issue looks related, #671

Did you check that one?

> that sounds like my vanilla quake sounds in my original pak should be fine

Is that the original pak file from the 90's, or the remastered version mentioned in #671?

**Comment by dsvensson** [comment]
It has nothing to do with CTF, it's just a github org I put the code under.

**Comment by ciscon** [comment]
fyi, certain (broken?) looping sounds don't work with the new sound system the way it is, if you compile with the old one they work.  the new one had a bug where these particular sounds would lock up the entire client so a fix was thrown in to stop that from happening, but it should actually be fixed properly by someone who knows how that thing actually works (eg @eb).

**Comment by jkr4m3r** [comment]
Yea, I'm the type to DL the client and play, not a coder, couldn't compile ezquake, or anything, if my life depended on it. Waiting for the "proper fix" is my only solution I guess. No big deal, just got tired of the guys using newer builds complaining about console errors when they joined the server.

We play more than a few outdoor maps too, so the ambient wind sounds etc are not a big loss.
